doc/designs/membermanager.md

```diff
@@ -0,0 +1,130 @@
+# Member Manager for group membership
+
+## Overview
+
+A member manager is a principal that is able to manage members of a
+group. Member managers are able to add new members to a group or remove
+existing members from a group. They cannot modify additional attributes
+of a group as a part of the member manager role.
+
+Member management is implemented for *user groups* and *host groups*.
+Membership can be managed by users or user groups. Member managers are
+independent from members. A principal can be a member manager of a
+group without being a member of a group.
+
+
+## Use Cases
+
+An administrator can use member management feature to delegate some
+control over user groups and host groups to users. For example a
+project manager is now able to add new team members to a project group.
+
+A NFS admin with member management capability for a host group is able
+to indirectly influence an HBAC rules and control which hosts can
+connect to an NFS file share.
+
+## Implementation
+
+The user group commands and host group commands are extended to handle
+member managers. The plugin classes grow two additional sub commands,
+one for adding and one for removing member managers. The show command
+prints member manager users and member manager groups. The find command
+can search by member manager.
+
+Member managers are stored in a new LDAP attribute ``memberManager``
+with OID 2.16.840.1.113730.3.8.3.28. It is multi-valued and contains
```

Comment:
As discussed on IRC, I moved the attribute into a new OID space for v4 attributes. Please use `2.16.840.1.113730.3.8.23.1.1` for `memberManager`